pkg/controller/tidbmonitor/tidb_monitor_control.go

```diff
@@ -14,36 +14,70 @@
 package tidbmonitor
 
 import (
+	"fmt"
+
 	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	v1alpha1validation "github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1/validation"
+	"github.com/pingcap/tidb-operator/pkg/client/clientset/versioned"
+	listers "github.com/pingcap/tidb-operator/pkg/client/listers/pingcap/v1alpha1"
 	"github.com/pingcap/tidb-operator/pkg/monitor"
-	"k8s.io/apimachinery/pkg/util/errors"
+	v1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	errorutils "k8s.io/apimachinery/pkg/util/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog"
 )
 
 // ControlInterface reconciles TidbMonitor
 type ControlInterface interface {
 	// ReconcileTidbMonitor implements the reconcile logic of TidbMonitor
 	ReconcileTidbMonitor(tm *v1alpha1.TidbMonitor) error
+
+	// Update tidbmonitor status
+	UpdateTidbMonitor(*v1alpha1.TidbMonitor) (*v1alpha1.TidbMonitor, error)
 }
 
 // NewDefaultTidbMonitorControl returns a new instance of the default TidbMonitor ControlInterface
-func NewDefaultTidbMonitorControl(monitorManager monitor.MonitorManager) ControlInterface {
-	return &defaultTidbMonitorControl{monitorManager: monitorManager}
+func NewDefaultTidbMonitorControl(cli versioned.Interface,
+	tmLister listers.TidbMonitorLister, monitorManager monitor.MonitorManager, recorder record.EventRecorder) ControlInterface {
+	return &defaultTidbMonitorControl{cli: cli, tmLister: tmLister, monitorManager: monitorManager, recorder: recorder}
```

Comment:
The `monitorManager.deps` includes all the `cli`s and `lister`s.

Reply:
monitorManager is interface , we can't use monitorManager.deps

Reply:
Do you mean deps as parameter directly?